File: asakatsu/251105/6/main.cs
static decimal Sqrt(decimal x, decimal epsilon = 0.00000000000001m)
{
    if (x == 0) return 0;
    decimal res = x / 2;
    while (Math.Abs(res * res - x) > epsilon)
    {
        res = (res + x / res) / 2;
    }
    return res;
}

decimal[] vars = Console.ReadLine().Split().Select(decimal.Parse).ToArray();
decimal x0 = vars[0];
decimal y0 = vars[1];
decimal r = vars[2];

decimal left = Math.Ceiling(x0 - r);
decimal right = Math.Floor(x0 + r);

long ans = 0;
for (decimal x = left; x <= right; ++x)
{
    decimal dx = x - x0;
    decimal dy = Sqrt(r * r - dx * dx);
    decimal bottom = Math.Ceiling(y0 - dy);
    decimal top = Math.Floor(y0 + dy);
    ans += (int)(top - bottom) + 1;
}
Console.WriteLine(ans);

File: asakatsu/251105/6/main.py
from decimal import *
from math import sqrt;

getcontext().prec = 21
x0, y0, r = map(Decimal, input().split())

left = int((x0 - r).quantize(Decimal("1"), rounding=ROUND_CEILING))
right = int((x0 + r).quantize(Decimal("1"), rounding=ROUND_FLOOR))

ans = 0
for x in range(left, right+1):
    dx = x - x0
    dy = (r * r - dx * dx).sqrt()
    bottom = (y0 - dy).quantize(Decimal("1"), rounding=ROUND_CEILING)
    top = (y0 + dy).quantize(Decimal("1"), rounding=ROUND_FLOOR)
    ans += int((top - bottom).to_integral_value()) + 1

print(ans)
